src/cc/table_desc.h

```diff
@@ -93,10 +81,10 @@ class TableDesc {
         leaf_snprintf(nullptr),
         is_shared(false),
         is_extern(false) {}
-  TableDesc(const std::string &name, FileDesc &&fd, int type, size_t key_size,
+  TableDesc(const std::string &name, int fd, int type, size_t key_size,
```

Comment:
This looks wrong to me, and was explicitly done this way initially. In particular, I don't want any possibility of invoking `FileDesc::operator int()` implicitly anywhere that a FileDesc must instead be move()'ed into the TableDesc. Such possibility would lead to double-close() of the underlying fd.

I don't have a problem with moving the class to common.h, but the `friend` and original implementation should remain.